### PR TITLE
[Merged by Bors] - feat(Topology, CategoryTheory): define light profinite sets

### PR DESCRIPTION
**Pull request by dagurtomas**
We define light profinite sets as structures containing the data of a sequential limit in `Profinite`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Category/LightProfinite/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  cone : Cone (diagram ⋙ FintypeCat.toProfinite.{u})
+  /-- The limit cone is limiting. -/
+  isLimit : IsLimit cone
+
+/-- A finite set can be regarded as a light profinite set as the limit of the constant diagram. -/
+def FintypeCat.toLightProfinite (X : FintypeCat) : LightProfinite where
+  diagram := (Functor.const _).obj X
+  cone := {
+    pt := toProfinite.obj X
+    π := eqToHom rfl }
```
Comment (joelriou) [inline Mathlib/Topology/Category/LightProfinite/Basic.lean:42]:
```suggestion
    π := Iso.refl _ }
```

File: Mathlib/Topology/Category/LightProfinite/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+end LightProfinite
+
+noncomputable section EssentiallySmall
+
+/-- This is an auxiliary definition used to show that `LightProfinite` is essentially small. -/
+structure LightProfinite' : Type u where
+  /-- The diagram takes values in a small category equivalent to `FintypeCat`. -/
+  diagram : ℕᵒᵖ ⥤ FintypeCat.Skeleton.{u}
+
+/-- A `LightProfinite'` yields a `LightProfinite`. -/
```
Comment (joelriou) [inline Mathlib/Topology/Category/LightProfinite/Basic.lean:109]:
```suggestion
/-- A `LightProfinite'` yields a `Profinite`. -/
```

File: Mathlib/Topology/Category/LightProfinite/Basic.lean [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+def LightProfinite'.toProfinite (S : LightProfinite') : Profinite :=
+  limit (S.diagram  ⋙ FintypeCat.Skeleton.equivalence.functor ⋙ FintypeCat.toProfinite.{u})
+
+instance : Category LightProfinite' := InducedCategory.category LightProfinite'.toProfinite
+
+/-- The functor part of the equivalence of categories `LightProfinite' ≌ LightProfinite`. -/
+def smallToLight : LightProfinite' ⥤ LightProfinite where
```
Comment (joelriou) [inline Mathlib/Topology/Category/LightProfinite/Basic.lean:116]: This definition should be moved to a suitable namespace:
```suggestion
def LightProfinite'.toLightFunctor : LightProfinite' ⥤ LightProfinite where
```

**Review by joelriou — APPROVED** [review]

**Comment by dagurtomas** [comment]
Thanks for the review, and in particular for pointing out the missing explicit universes!

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**